packages/subgraph/src/mappings/helpers.ts

```diff
@@ -225,3 +99,58 @@ export function updateHomeTokenInfo(
     log.debug('New overridden homeToken {}', [token.homeAddress.toHexString()]);
   }
 }
+
+// headerLength = 79 + sourceChainIdLength + destinationChainIdLength
+// for bsc, sokol, kovan, xdai and mainnet chainId < 255
+// => len(chainId) = 1
+var HEADER_LENGTH = 79 + 1 + 1;
+var METHOD_SIGNATURE_LENGTH = 4;
+var PADDED_LENGTH = 32;
+var ADDRESS_LENGTH = 20;
+
+var handleNativeTokensAndCall = Bytes.fromHexString('0x867f7a4d') as Bytes;
+var handleNativeTokens = Bytes.fromHexString('0x272255bb') as Bytes;
+var handleBridgedTokensAndCall = Bytes.fromHexString('0xc5345761') as Bytes;
+var handleBridgedTokens = Bytes.fromHexString('0x125e4cfb') as Bytes;
+var deployAndHandleBridgedTokensAndCall = Bytes.fromHexString(
+  '0xd522cfd7',
+) as Bytes;
+var deployAndHandleBridgedTokens = Bytes.fromHexString('0x2ae87cdd') as Bytes;
+
+export function decodeRecipient(encodedData: Bytes): Bytes {
```

Comment:
@akolotov I've tested this and it works. But please confirm if the logic looks right!